src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.OSX.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Globalization
+{
+    public partial class CompareInfo
+    {
+        private unsafe int CompareStringNative(ReadOnlySpan<char> string1, ReadOnlySpan<char> string2, CompareOptions options)
+        {
+            Debug.Assert(!GlobalizationMode.Invariant);
+            Debug.Assert(!GlobalizationMode.UseNls);
+            Debug.Assert((options & (CompareOptions.Ordinal | CompareOptions.OrdinalIgnoreCase)) == 0);
```

Comment:
You're checking for Ordinal here, so how did the test cases change the behavior?

Reply:
Why are we asserting that `Ordinal` and `OrdinalIgnoreCase` are not set? Any reason we are asserting that those two are not set but not having an assert for `StringSort`?

Reply:
Test cases are not changing the behaviour. I mistakenly skipped them for hybrid osx.
@mdh1418 as @ilonatommy mentioned in above comments CompareOptions.Ordinal and CompareOptions.OrdinalIgnoreCase are invariant algoritms and do not use ICU.

Reply:
@mdh1418 is right about early throwing, could we have something like:
https://github.com/dotnet/runtime/blob/f1819bd78f0e7506b8c3c7f90b47aa41b0e22c6f/src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.WebAssembly.cs#L41
for OSX? This way you spare time sending strings back and forth when the native function will reply with -2 anyway.

Reply:
Yes that sounds better, will add.

Reply:
This debug assert is replaced by the `AssertComparisonSupported`, we can remove it now.